src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -100,10 +101,24 @@ internal sealed partial class ObjectReader : IDisposable
                 return null;
             }
 
-            if (stream.ReadByte() != VersionByte1 ||
-                stream.ReadByte() != VersionByte2)
+            try
             {
-                return null;
+                if (stream.ReadByte() != VersionByte1 ||
+                    stream.ReadByte() != VersionByte2)
```

Comment:
Are the exceptions being thrown from these first calls to `ReadByte()` or perhaps later?

Reply:
➡️ Performance traces indicate that these are being thrown from inside `ReadByte` implementation.